tests/checkasm/vvc_sad.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2024 Stone Chen
+ *
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with FFmpeg; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <string.h>
+
+#include "checkasm.h"
+#include "libavcodec/vvc/ctu.h"
+#include "libavcodec/vvc/data.h"
+#include "libavcodec/vvc/dsp.h"
+
+#include "libavutil/common.h"
+#include "libavutil/intreadwrite.h"
+#include "libavutil/mem_internal.h"
+
+static const uint32_t pixel_mask[] = { 0xffffffff, 0x03ff03ff, 0x0fff0fff };
+
+#define SIZEOF_PIXEL ((bit_depth + 7) / 8)
+
+#define randomize_buffers(buf0, buf1, size, mask)           \
+    do {                                                    \
+        int k;                                              \
+        for (k = 0; k < size; k += 2) {                     \
+            uint32_t r = rnd() & mask;                      \
+            AV_WN32A(buf0 + k,     r);                      \
+            AV_WN32A(buf1 + k, r);                          \
+        }                                                   \
+    } while (0)
+
+#define randomize_pixels(buf0, buf1, size)                  \
+    do {                                                    \
+        uint32_t mask = pixel_mask[(bit_depth - 8) >> 1];   \
+        randomize_buffers(buf0, buf1, size, mask);          \
+    } while (0)
+
+
+static void check_vvc_sad_8_16bpc(void)
```

Comment:
similar functions,  could we combine them and put entire file to vvc_mc.c

Reply:
I've moved the checkasms into 1 function in vvc_mc.c
https://github.com/ffvvc/FFmpeg/blob/2ee95457a35b0701716ed17d27c967163c1ec3ea/tests/checkasm/vvc_mc.c#L325